src/test/java/com/epam/ta/tests/PostsTest.java

```diff
@@ -1,45 +1,141 @@
-import org.junit.Assert;
-import org.junit.jupiter.api.BeforeEach;
+import com.epam.ta.steps.PostSteps;
+import com.google.gson.Gson;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.http.HttpStatus;
+import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 
+import java.util.Arrays;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Slf4j
+@DisplayName("Verify Posts")
 public class PostsTest {
 
-    @BeforeEach
-    public void initTest() {
-        RestAssured.baseURI = "http://jsonplaceholder.typicode.com";
+    private static int TEST_POST_ID = 70;
+
+    private static int TEST_USER_ID = 7;
+
+    @Test
+    @DisplayName("Get all Posts")
+    public void verifyAllPosts() {
+        var response = PostEndpoint.getPosts();
+
+        assertThat(response.getStatusCode())
+                .as("wrong status code")
+                .isEqualTo(HttpStatus.SC_OK);
+
+        var posts = Arrays.asList(new Gson().fromJson(response.jsonPath().prettify(), Post[].class));
+        assertThat(posts.size()).isEqualTo(100);
+
+    }
+
+    @Test
+    @DisplayName("Get one Post")
+    public void verifyPost() {
+        var response = PostEndpoint.getPost(TEST_POST_ID);
+
+        assertThat(response.getStatusCode())
+                .as("wrong status code")
+                .isEqualTo(HttpStatus.SC_OK);
+
+        var post = new Gson().fromJson(response.jsonPath().prettify(), Post.class);
+        PostSteps.verifyPost(post, TEST_POST_ID, TEST_USER_ID);
+    }
+
+    @Test
+    @DisplayName("Get all comments for Post")
+    public void verifyPostComments() {
+        var response = PostEndpoint.getPostComments(TEST_POST_ID);
+
+        assertThat(response.getStatusCode())
+                .as("wrong status code")
+                .isEqualTo(HttpStatus.SC_OK);
+
+        var comments = Arrays.asList(new Gson().fromJson(response.jsonPath().prettify(), Comment[].class));
+        assertThat(comments.size()).isEqualTo(5);
+    }
+
+    @Test
+    @DisplayName("Get all Posts filter by UserId")
+    public void verifyPostsFilterByUserId() {
+        var response = PostEndpoint.getPostsFilterByUserId(TEST_USER_ID);
+
+        assertThat(response.getStatusCode())
+                .as("wrong status code")
+                .isEqualTo(HttpStatus.SC_OK);
+
+        var posts = Arrays.asList(new Gson().fromJson(response.jsonPath().prettify(), Post[].class));
+        assertThat(posts.size()).isEqualTo(10);
     }
 
     @Test
-    public void checkStatusCode() {
-        Response response = RestAssured.when()
-                .get("/posts")
-                .andReturn();
-        Assert.assertEquals(response.getStatusCode(), 200);
+    @DisplayName("Add new Post")
+    public void verifyAddPost() {
+        var testTitle = "title#1";
+        var testBody = "body#1";
+
+        var response = PostEndpoint.addPost(TEST_USER_ID, testTitle, testBody);
+
+        assertThat(response.getStatusCode())
```

Comment:
this assertion can be moved to some base class as it can be easiely shared between test classes